addons/point_of_sale/models/pos_session.py

```diff
@@ -686,7 +686,13 @@ def _post_statement_difference(self, amount, is_opening):
                 if not is_opening:
                     st_line_vals['counterpart_account_id'] = self.cash_journal_id.profit_account_id.id
 
-            self.env['account.bank.statement.line'].create(st_line_vals)
+            created_line = self.env['account.bank.statement.line'].create(st_line_vals)
+
+            if created_line:
+                session_link = "<a href='#' data-oe-model='pos.session' data-oe-id='%s'>%s</a>" % (self.id, self.name)
+                message = _("Related Session: %(link)s", link=Markup(session_link))
+                statement = self.env['account.move'].search([('statement_line_id', '=', created_line.id)])
+                statement.message_post(body=message)
```

Comment:
Nitpicking, I find it clearer to explicitly specify your content (and also allows to detect mistakes in case we change `session_link` later).

```suggestion
                statement = self.env['account.move'].search([('statement_line_id', '=', created_line.id)])
                statement.message_post(body=_(
                    "Related Session: %(link)s",
                    link=Markup("<a href='#' data-oe-model='pos.session' data-oe-id='%s'>%s</a>") % (self.id, self.name)
                ))
```